src/java.base/macosx/native/libjli/java_md_macosx.m

```diff
@@ -210,6 +210,8 @@
         preferredJVM = "client";
```

Comment:
#if defined(__i386__)
        preferredJVM = "client";
Not your bug, but Oracle/OpenJDK never supported 32-bit __i386__ on macOS.